client/src/app/pages/applications/applications-table/components/manage-columns-toolbar.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  Button,
+  OverflowMenu,
+  OverflowMenuGroup,
+  OverflowMenuItem,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { ColumnState } from "@app/hooks/table-controls/column/useColumnState";
+import { ManageColumnsModal } from "./manage-columns-modal";
+
+// Define props to accept columns and setColumns directly
+interface ManageColumnsToolbarProps<TColumnKey extends string> {
+  columns: ColumnState<TColumnKey>[];
+  setColumns: (newColumns: ColumnState<TColumnKey>[]) => void;
+}
+
+export const ManageColumnsToolbar = <TColumnKey extends string>({
+  columns,
+  setColumns,
+}: ManageColumnsToolbarProps<TColumnKey>) => {
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <>
+      <ToolbarItem variant="overflow-menu">
+        <OverflowMenu breakpoint="md">
+          <OverflowMenuGroup groupType="button" isPersistent>
+            <OverflowMenuItem isPersistent>
+              <Button variant="link" onClick={() => setIsOpen(true)}>
+                Manage columns
+              </Button>
+            </OverflowMenuItem>
+          </OverflowMenuGroup>
+        </OverflowMenu>
+      </ToolbarItem>
+
+      <ManageColumnsModal
```

Comment:
But since the modal is rendered on the table page for the page's configuration I'm not even sure what the state reset is for.  Adding the column management to any other table page will require the modal to be added there as well. :thinking: 

That doesn't invalidate the technique, it just make me curious.

Reply:
the modal keeps tracks of the columns in the internal state i.e.
```
const [editedColumns, setEditedColumns] =
    useState<ColumnState<TColumnKey>[]>(columns);
```
Most of the times this should be the same as the external state (maintained by the `useLocalTableControls()`) but it's safer to keep one source of truth.